src/pydvl/utils/types.py

```diff
@@ -64,3 +74,19 @@ def wrapper(*args, **kwargs):
         return fun(*args, **kwargs)
 
     return wrapper
+
+
+Seed = Union[int, np.random.SeedSequence]
+SeedOrGenerator = Union[Seed, np.random.Generator]
+
+
+def ensure_seedsequence(seed: Seed) -> Optional[SeedSequence]:
+    """Check if the seed is valid and return a SeedSequence object if it is. If it is
+    not valid, return None."""
+
+    if seed is None:
+        return None
+    elif isinstance(seed, int):
+        return SeedSequence(seed)
+    else:
+        return seed
```

Comment:
This could be the implementation for what I described in my other comment.

```suggestion
def ensure_seedsequence(seed: Optional[int, Generator, SeedSequence] = None) -> SeedSequence:
    """If the passed seed is a SeedSequence object then it is returned as is.
    Otherwise, a new object is created."""
    if isinstance(seed, SeedSequence):
        return seed
    elif isinstance(seed, Generator):
        return seed.bit_generator._seed_seq
    else:
        return SeedSequence(seed)
```

Reply:
See https://github.com/appliedAI-Initiative/pyDVL/pull/396#discussion_r1309621511